src/ShopRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Shopware\ServiceBundle;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Generator;
+use Shopware\ServiceBundle\Entity\Shop;
+
+class ShopRepository
+{
+    public function __construct(private readonly ManagerRegistry $registry) {}
+
+    /**
+     * @return Generator<Shop>
+     */
+    public function findAll(): Generator
```

Comment:
relevant to the comment on the command, I wonder if this can be rewritten to not fetch all shops then filter them, rather than pass the filter in the Doctrine 'findBy' criteria.

Reply:
refactored it now and introduced the method you mentioned